python/uds.py

```diff
@@ -473,16 +473,23 @@ def _isotp_rx_next(self, rx_data: bytes) -> ISOTP_FRAME_TYPE:
     # assert len(rx_data) == self.max_len, f"isotp - rx: invalid CAN frame length: {len(rx_data)}"
 
     if rx_data[0] >> 4 == ISOTP_FRAME_TYPE.SINGLE:
-      self.rx_len = rx_data[0] & 0xFF
-      assert self.rx_len < self.max_len, f"isotp - rx: invalid single frame length: {self.rx_len}"
-      self.rx_dat = rx_data[1:1 + self.rx_len]
+      self.rx_len = rx_data[0] & 0x0F
+      # "if the first byte of SF=0, then second byte specifies the size of the data."
+      # - https://en.wikipedia.org/wiki/CAN_FD
+      offset = 1
+      if self.rx_len == 0x0 and len(rx_data) > 8:
```

Comment:
```suggestion
      if self.rx_len == 0x0 and len(rx_data) >= rx_data[1]:
```

Reply:
Worth noting we did have some issues with this. @sheaduncan can probably explain more.

Reply:
Which version?

Reply:
Instead of using a hardcoded 8, using the `rx_data[1]` caused it to fail to fingerprint properly for him. I dont' have much more info other than that. I've used the `> 8` code since the beginning and that's seemingly working with no issues.